### DISCUSSION TITLE
Is it possible to install act on linux, macos and use them from remote in LAN?

### DISCUSSION THREAD

**Discussion by Dexus** [nektos/act#2592] [Q&A]
I have multiple Servers for DEV, now I'm asking me, if it is possible to run act on Linux and MacOS and use it from remote - but with cli on my Windows DEV Maschine?

**Comment by ChristopherHX** [comment]
Happy new year 🎆 !

> if it is possible to run act on Linux /.../ and use it from remote - but with cli on my Windows DEV Maschine?

Yes, `DOCKER_HOST=ssh://linux-server` with docker engine (* act still runs on windows, but docker executes commands on your linux server instead of a local docker desktop VM)

> if it is possible to run act on /.../ macOS and use it from remote - but with cli on my Windows DEV Maschine?

No, act has no remote protocol for `-P macos-latest=-self-hosted`. Of course linux workload can work like on linux, but you imply to me this was not your question.

_For more complex scenarious if and only if act is not a requirement, e.g. using a server to distribute workload with a partially act compatible cli you can look at my actions/runner fork (not mentioning the name is my intention due to information embargo). I have a github-act-runner executor as well that is build around my fork of act, but the official actions/runner is better tested for running the things on the server_

I'm linking your posts now https://github.com/SanjulaGanepola/github-local-actions/issues/131

**Reply by Aetherinox** [reply]
Just to confirm your state of there being no remote protocol.

I would assume the following is impossible right now:

- I have a github runner hosted on another server, off-site.
- Instead of using it with Github, Id' like to use it with act since it has the image and all the packages I need.

I'm guessing at the moment, there's no way to specify a remote github self-hosted runner and use that correct? Or would I have to bring the runner over the local machine that is running act?

**Reply by ChristopherHX** [reply]
yes this is correct, there's no way to specify a remote github self-hosted runner and use that for nektos/act.

- You would have to ssh into your self-hosted runner (if possible, otherwise tmate action could be used from a github workflow job that has a some input lag).

- install act on the self-hosted you sshed into
- Use `-P my-self-hosted-label=-self-hosted`, to run everything on your self-hosted runner. This could fail, depending on the installed tool due to a missing `node` nodejs program.

- Then work inside that self-hosted runner image using the vscode-ssh remote like a github codespace or cli interface.

**Reply by Aetherinox** [reply]
Thanks, just wanted to confirm before I go spending time trying to do something I can't. The default runner functionality is fine and it works, the only downside is that in order to run all the jobs in my workflow, I had to install `-P ubuntu-latest=catthehacker/ubuntu:full-22.04`. And that image is like 60+GB.

When I tried the smaller images, it threw errors about not being able to define the date, such as

```
echo "NOW=$(date +'%m-%d-%Y %H:%M:%S')" >> $GITHUB_ENV  
```

And another image wouldn't let `actions/github-script` to run. Once I installed the larger image, it worked fine. Just 60GB is quite the image size. I either have to let it sit so I can use it later, or I re-download 60GB every time I need to test a workflow. And even on a 700mbps / down, 60GB still takes a bit to download. (roughly 11 minutes)
